### ISSUE TITLE
Vesktop hangs when launched through Steam

### ISSUE THREAD

**Issue by e-tho** [Vencord/Vesktop#398]
**Describe the bug**

Basically the same issue as #182

**To Reproduce**

1. Open Steam
2. Click on "Add a game" in the bottom left-hand corner of the Steam window.
3. Click on "Add a Non-Steam Game" and select Vesktop from the applications list.
4. Launch Vesktop from Steam

**Desktop (please complete the following information):**
 - OS/Distro: NixOS
 - Desktop Environment: Hyprland

**Additional context**

I've added emulators as non-Steam games and they launch correctly, it doesn't seem to be due to my environment.
I haven't seen any particular error/warning message, the launch just hangs for a while.
I'm on the NixOS unstable channel, so the Vesktop package version is the current one: 1.5.0.
I've tried adding the environment variable `ELECTRON_OZONE_PLATFORM_HINT=auto` (available as of electron 28) without success.

**Comment by Vendicated** [comment]
does it work when launching without steam?

why do you want to launch it through steam? we have steam specific code but it expects you to be using a steamdeck because running through steam in any other case makes no sense

**Comment by e-tho** [comment]
It works perfectly when I run it without steam.
I have a Gamescope embedded session (Adaptive Sync only works in this mode, not as nested) running Steam Big Picture and I'd like to be able to access Vesktop from that session, it's the only way I've found so far.

**Comment by Vendicated** [comment]
well i highly doubt anyone will be working on this because it's very niche. i suggest you try to fix it yourself. i checked again and the steamos fixes are only applied if its properly steamos, so you could try tricking vesktop into thinking you're on steamos by setting these env vars

https://github.com/Vencord/Vesktop/blob/main/src/main/utils/steamOS.ts#L22

**Comment by italoghost** [comment]
Hi! I am using an gamescope session as well on my machine and applying the following environment variables worked:

`env SteamOS=1 SteamGamepadUI=1vesktop`

@Vendicated I wonder if you could add it to the README to users that want to launch Vesktop through a gamescope session.

**Comment by e-tho** [comment]
I had already tried this solution without success, in my case setting the environment variables via the Steam launch options
```
SteamOS=1 SteamGamepadUI=1 %command%
```
 or via the NixOS module
```
programs.steam = {
  enable = true;
  gamescopeSession = {
    enable = true;
    env = {
      SteamOS = "1";
      SteamGamepadUI = "1";
    };
  };
};
```
had no effect.

**Comment by AAGaming00** [comment]
what GPU vendor & where is the source of your gamescope session launch script

**Comment by italoghost** [comment]
> I had already tried this solution without success, in my case setting the environment variables via the Steam launch options
> 
> ```
> SteamOS=1 SteamGamepadUI=1 %command%
> ```
> 
> or via the NixOS module
> 
> ```
> programs.steam = {
>   enable = true;
>   gamescopeSession = {
>     enable = true;
>     env = {
>       SteamOS = "1";
>       SteamGamepadUI = "1";
>     };
>   };
> };
> ```
> 
> had no effect.

You have to put it before the binary: 

![image](https://github.com/Vencord/Vesktop/assets/48540550/ac714453-7716-415b-b162-aaf48c67050b)

**Comment by italoghost** [comment]
> what GPU vendor & where is the source of your gamescope session launch script

I am using [gamescope-session-git](https://aur.archlinux.org/packages/gamescope-session-git) and [gamescope-session-steam-git](https://aur.archlinux.org/packages/gamescope-session-steam-git) from ChimeraOS, on my ArchLinux installation.

**Comment by AAGaming00** [comment]
> what GPU vendor

interesting though, @Vendicated maybe for gamescope on Linux we could enable a subset of the steamOS stuff?

**Comment by Vendicated** [comment]
yes sounds good

**Comment by italoghost** [comment]
I don't know what has changed but I had to add "--no-sandbox" as a argument to it to work today with the environment variables

**Comment by e-tho** [comment]
Indeed, with `--no-sandbox` Vesktop launches correctly via Steam, @italoghost my version must have been more recent than yours because the environment variables were not sufficient in my case. Thanks for the workaround.

**Comment by Vendicated** [comment]
Vesktop already adds this command line flag if the SteamOS env vars are set:
```sh
if [[ "$SteamOS" == "1" && "$SteamGamepadUI" == "1" ]]; then
    echo "Running Vesktop on SteamOS, disabling sandbox"
    IS_STEAMOS=1
fi
```

So I think this issue is resolved? If you think this should be documented, you can add it to https://vesktop.dev/linux/tips/ - just click the edit button on the bottom
